lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
+        Dir.glob(File.join(Chef::Util::PathHelper.escape_glob(cookbook_path), "*")).each do |dir|
```

Comment:
you should use escape_glob_dir here instead of escape_glob (check the pull request that added escape_glob_dir for the long story...)